Week01/Problem04/wyou_04.py

```diff
@@ -0,0 +1,2 @@
+large = max(a*b for a in range(101,1000) for b in range(101,1000) if str(a*b) == str(a*b)[::-1])
```

Comment:
This script, too, would benefit from comments.  Also, while list comprehensions are cool and concise, they occasionally can become too complex and difficult to read.  For example, this single line could be expanded to look something like this:
```
# Make empty list for palindromes
palindromes = []
# Loop over possible values for product
for a in range(101, 1000):
    for b in range(101, 1000):
        # Is the product a palindrome? (same forwards & backwards)
        if str(a*b) == str(a*b)[::-1]:
            # Append palindrome to list
            palindromes.append(a*b)

# Get largest palindrome from list
large = max(palindromes)
print(large)
```
which would be much more clear for someone (including yourself!) reading your code six months from now.